### PR TITLE
Issue 9807 backend create recent content widgets for the add new content functionality

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: src/com/dotcms/rest/api/v1/content/ContentTypeHelper.java [hunk 1]
```diff
@@ -86,34 +88,65 @@ protected ContentTypeHelper(WebResource webResource,
...
 
         return result;
     }
 
+    private void addRecents(final HttpServletRequest request, final User user, Structure.Type type,
+                                            List<BaseContentTypesView>  baseContentTypesView)
+            throws DotDataException, LanguageException {
+
+        Locale locale = LocaleUtil.getLocale(request);
+        Map<String, String> baseContentTypeNames = this.getBaseContentTypeNames(locale);
```
Comment (jdotcms) [inline src/com/dotcms/rest/api/v1/content/ContentTypeHelper.java]: This should be thread-safe and the map should be inmutable

File: src/com/dotcms/rest/api/v1/content/ContentTypeHelper.java [hunk 2]
```diff
@@ -86,34 +88,65 @@ protected ContentTypeHelper(WebResource webResource,
...
+            map = map(
+                    Structure.Type.CONTENT.name(), LanguageUtil.get(locale, "Content"),
+                    Structure.Type.WIDGET.name(), LanguageUtil.get(locale, "Widget"),
+                    Structure.Type.FORM.name(), LanguageUtil.get(locale, "Form"),
+                    Structure.Type.FILEASSET.name(), LanguageUtil.get(locale, "File"),
+                    Structure.Type.HTMLPAGE.name(), LanguageUtil.get(locale, "HTMLPage"),
+                    Structure.Type.PERSONA.name(), LanguageUtil.get(locale, "Persona")
+            );
+
+            BASE_CONTENT_TYPE_LABELS.put(locale, map);
```
Comment (jdotcms) [inline src/com/dotcms/rest/api/v1/content/ContentTypeHelper.java]: This should be inmutable

File: src/com/dotmarketing/cache/ContentTypeCacheImpl.java [hunk 3]
```diff
@@ -22,8 +26,10 @@
     private final String primaryGroup = "StructureCache";
     private final String containerStructureGroup = "ContainerStructureCache";
     private final String structuresByTypeGroup = "StructuresByTypeCache";
+    private final String recentGroup = "RecentCache";
```
Comment (jdotcms) [inline src/com/dotmarketing/cache/ContentTypeCacheImpl.java]: Should be this a constant?

**Review by jdotcms — CHANGES_REQUESTED** [review]

File: dotCMS/WEB-INF/messages/Language.properties [hunk 4]
```diff
@@ -910,6 +911,7 @@ random=random
 Verify-Password=Verify Password
 Content=Content
 Widget=Widget
+recent_widget=Recent Widget
```
Comment (jgambarios) [inline dotCMS/WEB-INF/messages/Language.properties]: I think a **s** is missing:
recent_widgets=Recent Widgets

**Review by jgambarios — CHANGES_REQUESTED** [review]

File: src/com/dotmarketing/cache/ContentTypeCacheImpl.java (hunk 3 above)
Comment (jgambarios) [inline src/com/dotmarketing/cache/ContentTypeCacheImpl.java]: @wezell that recent cache is where we are planning to store the recently used content types.

**Review by jgambarios — APPROVED** [review]

**Pull request merged**